### ISSUE TITLE
Custom rule regex with lookbehind to detect files missing a copyright?

### ISSUE THREAD

**Issue by heidiproske** [realm/SwiftLint#2946]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### (New rule) request for help
I'm trying to write a custom rule to help detect files that are missing a certain form of copyright: `Copyright © \\d{4} CompanyX\\. All rights reserved\\.`

I originally practiced my regex using a "negative lookahead" but based on the SwiftLint source code, it seems like only [negative lookbehinds are supported]( https://github.com/realm/SwiftLint/blob/master/Source/SwiftLintFramework/Rules/Lint/YodaConditionRule.swift#L10).

My attempt in Xcode seems to do the right thing using this regex: `(?s)\A((?<!Copyright).)+\z`

<img width="849" alt="regexForCopyright" src="https://user-images.githubusercontent.com/6706831/68347412-b556e580-00ab-11ea-97c9-8d44b380f928.png">

Yet when I try to convert this to add to my `.swiftlint.yml` (by escaping my backslashes), it doesn't trigger the problem files:
```
custom_rules:
  require_copyright:
    included: ".*\\.swift"
    name: "Missing Copyright"
    regex: "(?s)\\A((?<!Copyright).)+\\z"
    message: "Source files should have a copyright disclaimer"
    severity: error
```

(PS: I did try escaping the `.` signalling each position to inspect at.)

Is anyone able to help me 🙏fix my regex or let me know if I missed an existing rule for this?

**Comment by marcelofabri** [comment]
You can use the `file_header` rule for this:

```yml
file_header:
  required_string: "Copyright"
```

It's an opt-in rule, so you also need to enable it in `opt_in_rules`.

**Comment by heidiproske** [comment]
@marcelofabri Ah thank you, sorry I missed that customization in the `file_header` rule. 

Is there a `YEAR` placeholder for `d{4}`, similar to the filename? 

i.e. we have different files with different years, so was wanting to do something like 
`required_string: "Copyright © YEAR CompanyX. All rights reserved.`

**Comment by marcelofabri** [comment]
No, but you can use `required_pattern` instead and provide a regex

**Comment by heidiproske** [comment]
Thanks @marcelofabri!
